java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -354,6 +355,18 @@ protected boolean compareStructVectors(Range range) {
       return false;
     }
 
+    if (leftVector instanceof StructVector || rightVector instanceof StructVector) {
```

Comment:
What if we do this check for all vectors not just `StructVector`?

Reply:
Another thing is this visitor interface is used for all vectors, and specifying a logic just for a single vector type doesn't make much sense. Further, if we are to check the range considering nulls, it should be for all vectors?

Reply:
There's already null compacrion for other types of vector.

Reply:
Oh, I misinterpreted.